### PR TITLE
[CR needed] Fourier tempogram, PLP and friends

### PR DESCRIPTION
**Pull request by bmcfee**
#### Reference Issue
Implements #739 

#### What does this implement/fix? Explain your changes.

This PR implements predominant local pulse `beat.plp`, Fourier tempogram `feature.fourier_tempogram`, helper unit converter `core.fourier_tempo_frequencies` and axis decoration support for Fourier tempograms.

#### Any other comments?

In implementing the unit tests for this, I discovered that the matplotlib image comparison tests were not executing properly (silently passing).  I've fixed that here too, and in the process regenerated the baseline images.

I think this is basically good to go.  I'm considering adding an advanced example notebook that shows how to use PLP with DP beat tracking, as done in section V.E of Grosche & Mueller 2011.

### CODE REVIEW

**Comment by bmcfee** [comment]
Test issues are now fixed.  I think this one's ready for CR.

**Comment by bmcfee** [comment]
@stefan-balke any chance you could cr this?  it's cool if not, just let me know.

File: librosa/beat.py [hunk 1]
```diff
@@ -342,6 +343,166 @@ def tempo(y=None, sr=22050, onset_envelope=None, hop_length=512, start_bpm=120,
...
+
+    # Step 3: Discard everything below the peak
+    ftmag = np.abs(ftgram)
+    peak_values = ftmag.max(axis=0, keepdims=True)
+    ftgram[ftmag < peak_values] = 0
+
+    # Normalize to keep only phase information
+    ftgram[:] /= peak_values
+
+    # Step 5: invert the fourier tempogram to get the pulse
```
Comment (lostanlen) [inline librosa/beat.py]: Fourier

File: librosa/core/time_frequency.py [hunk 2]
```diff
@@ -1085,6 +1086,40 @@ def tempo_frequencies(n_bins, hop_length=512, sr=22050):
     return bin_frequencies
 
 
+def fourier_tempo_frequencies(sr=22050, win_length=384, hop_length=512):
+    '''Compute the frequencies (in beats-per-minute) corresponding
```
Comment (lostanlen) [inline librosa/core/time_frequency.py]: beat per minute (no hyphens)

File: librosa/feature/rhythm.py [hunk 3]
```diff
@@ -177,3 +179,109 @@ def tempogram(y=None, sr=22050, onset_envelope=None, hop_length=512,
...
+
+    Examples
+    --------
+    >>> # Compute local onset autocorrelation
+    >>> y, sr = librosa.load(librosa.util.example_audio_file())
+    >>> hop_length = 512
+    >>> oenv = librosa.onset.onset_strength(y=y, sr=sr, hop_length=hop_length)
+    >>> tempogram = librosa.feature.fourier_tempogram(onset_envelope=oenv, sr=sr,
+    ...                                               hop_length=hop_length)
+    >>> # compute the auto-correlation tempogram, unnormalized to make comparison easier
```
Comment (lostanlen) [inline librosa/feature/rhythm.py]: Compute (uppercase)

File: librosa/display.py [hunk 4]
```diff
@@ -977,6 +983,21 @@ def __coord_tempo(n, sr=22050, hop_length=512, **_kwargs):
     return basis * (edges + 0.5) / edges
 
 
+def __coord_fourier_tempo(n, sr=22050, hop_length=512, **_kwargs):
+    '''Fourier tempogram coordinates'''
+
+    n_fft = 2 * (n - 1)
+    # The following code centers the FFT bins at their frequencies
+    # and clips to the non-negative frequency range [0, nyquist]
```
Comment (lostanlen) [inline librosa/display.py]: nonnegative

**Comment by bmcfee** [comment]
> LGTM. I pointed out a few cosmetic issues in the docstring

Thanks!  I'll clean those up and get it pushed.

File: librosa/beat.py [hunk 5]
```diff
@@ -342,6 +343,166 @@ def tempo(y=None, sr=22050, onset_envelope=None, hop_length=512, start_bpm=120,
...
+                               sr=sr, hop_length=hop_length,
+                               win_length=win_length)
+
+    # Step 3: pin to the feasible tempo range
+
+    tempo_frequencies = core.fourier_tempo_frequencies(sr=sr,
+                                                       hop_length=hop_length,
+                                                       win_length=win_length)
+
+    # TODO: in the future, generalize this to support arbitrary priors
```
[the remark this answers is not in the document]
Comment (bmcfee) [reply librosa/beat.py]: I'll fix that in the future PR for #880

File: librosa/display.py (hunk 4 above)
Comment (bmcfee) [reply librosa/display.py]: we use non-negative elsewhere in the code, so I'd rather keep as is.

**Comment by bmcfee** [comment]
:shipit: thanks for the CR!

**Comment by stefan-balke** [comment]
thanks @lostanlen and sry for not being responsive. Pretty busy these days...

**Pull request merged**